jOOQ/src/main/java/org/jooq/impl/InsertQueryImpl.java

```diff
@@ -278,10 +279,21 @@ else if (onDuplicateKeyIgnore) {
                 case POSTGRES_9_5:
                 case POSTGRES: {
                     toSQLInsert(ctx);
-                    ctx.formatSeparator()
-                       .start(INSERT_ON_DUPLICATE_KEY_UPDATE)
-                       .keyword("on conflict do nothing")
-                       .end(INSERT_ON_DUPLICATE_KEY_UPDATE);
+                    ctx.formatSeparator().start(INSERT_ON_DUPLICATE_KEY_UPDATE).keyword("on conflict");
+
+                    if (onConflict != null && onConflict.size() > 0) {
+                        if (onConflict.size() != 1) {
+                            LOGGER.info(
+                                    "ON DUPLICATE KEY IGNORE accept only one field for unique/exclusion constraint, " +
+                                            "let's take the first one on the list {}", onConflict.get(0).getName());
+                        }
```

Comment:
Is this really correct? Can you point me to the docs where this is specified? I would imagine that we should be able to list the columns of a composite key here...